alica_tests/src/test/test_alica_dynamic_task.cpp

```diff
@@ -0,0 +1,149 @@
+#include "test_alica.h"
+
+#include "DynamicTaskBehavior.h"
+
+#include <alica/test/Util.h>
+#include <engine/AlicaContext.h>
+#include <engine/RunningPlan.h>
+#include <engine/model/AlicaElement.h>
+#include <engine/model/Behaviour.h>
+#include <engine/model/EntryPoint.h>
+#include <engine/model/Plan.h>
+#include <engine/model/State.h>
+#include <engine/model/Task.h>
+
+#include <gtest/gtest.h>
+
+namespace alica
+{
+namespace
+{
+// GTEST_FILTER=AlicaDynamicTaskPlan.* catkin run_tests -i alica_tests
+class AlicaDynamicTaskPlan : public AlicaTestMultiAgentFixture
+{
+protected:
+    const char* getRoleSetName() const override { return "Roleset"; }
+    const char* getMasterPlanName() const override { return "DynamicTaskAssignmentTestMaster"; }
+
+    const int agentCount = 2;
+    int getAgentCount() const override { return agentCount; }
+    const char* getHostName(int agentNumber) const override
+    {
+        if (agentNumber) {
+            return "hairy";
+        } else {
+            return "nase";
+        }
+    }
+
+    void startAgents()
+    {
+        for (uint8_t agent_index = 0; agent_index < getAgentCount(); agent_index++) {
+            aes[agent_index]->start();
+        }
+    }
+
+    void stepAgents()
+    {
+        for (uint8_t agent_index = 0; agent_index < getAgentCount(); agent_index++) {
+            acs[agent_index]->stepEngine();
+        }
+    }
+
+    void enableTransitionCondition()
+    {
+        // do we need to GetOne and GetTwo? :O
+        alicaTests::TestWorldModel::getOne()->setTransitionCondition4496654201854254411(true);
+        alicaTests::TestWorldModel::getTwo()->setTransitionCondition4496654201854254411(true);
+    }
+
+    static void checkAlicaElement(const alica::AlicaElement* ae, long id, std::string name)
+    {
+        EXPECT_EQ(id, ae->getId()) << "Wrong ID!" << std::endl;
+        EXPECT_STREQ(name.c_str(), ae->getName().c_str()) << "Wrong Name for AlicaElement" << std::endl;
```

Comment:
std::endl isn't necessary I think.  gtest already outputs a new line after the error message unless you want to add an extra one.  I also think most of the comments like this don't really add anything since the error msg already makes it clear which things are being compared.  It only helps in cases where the variable names suck, and I would prefer improving that (i.e. `alicaElement` rather than `ae`